docs/task.md

```diff
@@ -0,0 +1,121 @@
+# Task
+
+Task is the smallest unit for running commands in SDK.
+It can be used as a single runnable console command or used in more complex structures such as the [workflows](workflow.md) or [task-sets](task_set.md).
+In another word task is a commands wrapper that makes them extensible, configurable, versionable and provides cli interface for them.
+
+## How to run task
+
+- `spryker-sdk <task-id>` to run task.
+- `spryker-sdk <task-id> --stages=stage_a --stages=stage_b` to run task commands filtered by stages.
+- `spryker-sdk <task-id> --tags=tag_a --tags=tag_b` to run task commands filtered by tags.
+- `spryker-sdk <task-id> --help` or `spryker-sdk <task-id> -h` to get task help info.
+
+## How to update tasks
+
+`sdk:update:all` - executes the updates of all existing tasks. The [lifecycle events](lifecycle_management.md) will be triggered after.
+
+`rm -rf ./db && spryker-sdk sdk:init:sdk` - the hard way (only for developers)
+
+## How to create task
+
+Task can be created in a declarative way by specifying the task yaml configuration file or by implementing `\SprykerSdk\SdkContracts\Entity\TaskInterface` as a php class.
+The declarative is preferred way unless more complex logic is needed.
+
+### Task yaml configuration reference
+
+Configuration file should be placed in `extension/*/config/task/` or `src/Extension/Resources/config/task/` directory.
+
+- `id` - __unique__ task id. Should consist only of `[\w\:]+` symbols.
```

Comment:
Please update all items with required or optional tags in the beginning. Example:
- `lifecycle` - (optional) - lifecycle commands list.

As an alternative way, you can change it to table view.